source/Calamari.Azure/Deployment/Conventions/DeployAzureResourceGroupConvention.cs

```diff
@@ -79,16 +78,17 @@ public void Install(RunningDeployment deployment)
             PollForCompletion(createArmClient, resourceGroupName, deploymentName, variables);
         }
 
-        static string GenerateDeploymentNameFromStepName(string stepName)
+        protected static string GenerateDeploymentNameFromStepName(string stepName)
         {
             var deploymentName = stepName ?? string.Empty;
             deploymentName = deploymentName.ToLower();
             deploymentName = Regex.Replace(deploymentName, "\\s", "-");
-            deploymentName =
-                new string(deploymentName.Select(x => (char.IsLetterOrDigit(x) || x == '-') ? x : '-').ToArray());
+            deploymentName = new string(deploymentName.Select(x => (char.IsLetterOrDigit(x) || x == '-') ? x : '-').ToArray());
             deploymentName = Regex.Replace(deploymentName, "-+", "-");
             deploymentName = deploymentName.Trim('-', '/');
-            deploymentName = deploymentName + "-" + Guid.NewGuid().ToString();
+            // Azure Deployment Namese can only be 64 characters == 27 chars + "-" (1) + Guid (36 chars)
+            deploymentName = deploymentName.Length <= 27 ? deploymentName : deploymentName.Substring(0, 27);
+            deploymentName = deploymentName + "-" + Guid.NewGuid();
```

Comment:
We could use `guid.ToString("N")` which would allow us to use a few more characters that were provided by the end user. We could also base the number of available characters on the length of the unique value.  Then we don't have to have the comment above.